prometheus/handler.go

```diff
@@ -0,0 +1,136 @@
+package prometheus
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/segmentio/stats"
+)
+
+// Handler is a type that bridges the stats API to a prometheus-compatible HTTP
+// endpoint.
+//
+// Typically, a program creates one Handler, registers it to the stats package,
+// and adds it to the muxer used by the application under the /metrics path.
+type Handler struct {
+	// A map of metric name to histograms buckets used by the handler to record
+	// observed values.
+	//
+	// The buckets are of list of upper limits used to group the observed
+	// values.
+	//
+	// If a bucket is missing for a metric name the observed value is ignored.
+	HistgramBuckets map[string][]float64
```

Comment:
That was what I thought initially as well but after reading more of the documentation on histograms in prometheus it seemed like they are much more flexible than summaries => https://prometheus.io/docs/practices/histograms/#quantiles

Basically the trade off is to be able to ask more questions about the data set but have the query computation be more expansive. Summaries are kind of "prepared queries" that answer only one question about the data set, and all the aggregation work is moved to the client side so the queries are basically free to run for the server.

The downside of histogram also appears to be addressable via the use of [recording rules](https://prometheus.io/docs/querying/rules/#recording-rules) so it appeared to be a much more powerful tool to implement.

We also have to be mindful of not limiting ourselves to the datadog model because we're familiar with it, this is a different tool and we should embrace the difference to get as much leverage as we can.

All that being said I'm open to revisiting the implementation if we realized it didn't fit what we're trying to measure.